File: Export/Interfaces/IExportService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace Export.Interfaces
{
    /// <summary>
    /// Interface for module that handles queuing and progress tracking of export jobs.
    /// </summary>
    public interface IExportService : INotifyPropertyChanged
    {
        /// <summary>
        /// Queue a job to be exported.
        /// </summary>
        /// <param name="export"></param>
        void QueueExportJob(IPdfExport export);

        /// <summary>
        /// Check whether the application is currently exporting something.
        /// </summary>
        bool ExportInProgress { get; set; }

        /// <summary>
        /// Get the percentage of export progress completed.
        /// </summary>
        int ExportProgress { get; set; }

        /// <summary>
        /// Get the name of document currently being exported, or null if none.
        /// </summary>
        string CurrentDocument { get; set; }

        /// <summary>
        /// Event that will be raised when all exports are completed.
        /// </summary>
        event EventHandler AllExportsCompleted;
    }
}

File: ExportF/IExportService.fs
namespace ExportF

open System
open System.Collections.Generic
open System.Linq
open System.Text
open System.ComponentModel

/// Interface for module that handles queuing and progress tracking of export jobs.
type IExportService = 
    /// Queue a job to be exported.
    abstract QueueExportJob : IPdfExport
    /// Check whether the application is currently exporting something.
    abstract member ExportInProgress : bool with get, set
    /// Get the percentage of export progress completed.
    abstract member ExportProgress : bool with get, set
    /// Get the name of document currently being exported, or null if none.
    abstract member CurrentDocument : string with get, set
    /// Event that will be raised when all exports are completed.
    abstract AllExportsCompleted : EventHandler
